### ISSUE TITLE
Add callback for knn-validation

### ISSUE THREAD

**Issue by philippmwirth** [lightly-ai/lightly#77]
Add hooks for validation at the end of each epoch (only informative if there are labels). Let's make sure it can be switched off and that the user can pass a validation set of his choice.

**Comment by busycalibrating** [comment]
I can handle this after I finish #61, or potentially before if we end up using this to reproduce some of the results (CIFAR-10) in the SimSiam paper

**Comment by philippmwirth** [comment]
Let's go one-by-one because we already have an internal version of this with which we can evaluate whether SimSiam works as expected.

**Comment by IgorSusmelj** [comment]
I don't like callbacks for that. It would just blow up the modules. Not everyone wants to run kNN-validation and needs this. You could only use it for image classification and only if you have labels. 
Why not adding some utility methods instead to quickly build one yourself and provide the sample code? That way it could be easier to adapt it.

**Comment by busycalibrating** [comment]
I would agree we shouldn't dump every little thing we think up into its own module, however I do think offering some basic callbacks would not be inappropriate. For example, PL has a callback for an [online evaluator](https://pytorch-lightning-bolts.readthedocs.io/en/stable/self_supervised_callbacks.html).
